stdlib/public/core/CTypes.swift

```diff
@@ -63,11 +63,26 @@ public typealias CFloat = Float
 public typealias CDouble = Double
 
 /// The C 'long double' type.
+#if os(OSX) || os(iOS) || os(watchOS) || os(tvOS)
+// On Darwin, long double is Float80 on x86, and Double otherwise.
 #if arch(x86_64) || arch(i386)
 public typealias CLongDouble = Float80
 #else
 public typealias CLongDouble = Double
 #endif
+#elseif os(Windows)
+// On Windows, long double is always Double.
+public typealias CLongDouble = Double
+#elseif os(Linux)
+// On Linux/x86, long double is Float80.
+// TODO: Fill in definitions for additional architectures as needed. IIRC
+// armv7 should map to Double, but arm64 and ppc64le should map to Float128,
```

Comment:
This is correct, I would drop the "IIRC".  ARMv7 Linux maps `long double` to `double` both of which are FP64.  ARM64 and PPC64 both use FP128.

Reply:
Hey, @compnerd, want to implement FP128? ;-)

Reply:
I might ... 🤦‍♂️

Reply:
We all *want* to implement `Float128`, it's just a matter of free time =)

Reply:
I'm pretty sure it's not going to be harder than Float80. It's all just forwarding to LLVM.

…well, okay, the fact that tgmath doesn't have overloads for it is kind of annoying.